src/weatherDataSource.js

```diff
@@ -0,0 +1,42 @@
+export default class WeatherDataSource {
+    constructor() {
+        this.apiKey = '84af29f03ef35bb4dd7f1e9b0ad575e2';
+    }  
+    getweather(city) { 
+        fetch('https://api.openweathermap.org/data/2.5/weather?q=' + city + '&appid=' + this.apiKey)  
+            .then(function(resp) { 
+                return resp.json() })
+            .then(function(data) {
+                console.log(data);
+                showWeather(data);
+            })
+            .catch(function(err) {
+                console.log(err);
+            });
+    }
+}
+const input = document.getElementById("submit");
+var city = "";
+var weather = new WeatherDataSource();
+input.addEventListener('click', function(e) {
+    e.preventDefault();
+    city = document.querySelector("#city").value;
+    weather.getweather(city);
+});
+
+//dodatkowa funkcja związana z pokazywaniem informacji na stronie
+function showWeather(data) {
+    var celcius = Math.round(parseFloat(data.main.temp)-273.15);
+    var description = data.weather[0].description;
+    
+    if( description.indexOf('rain') > 0 ) {
+        document.body.className = 'rainy';
+    } else if( description.indexOf('cloud') > 0 ) {
+        document.body.className = 'cloudy';
+    } else if( description.indexOf('sunny') > 0 ) {
+        document.body.className = 'sunny';
+    }
+    
+    document.getElementById('description').innerHTML = description;
+    document.getElementById('temp').innerHTML = celcius + '&deg;';
+}
```

Comment:
This shouldn't be part of weatherDataSource.js file. Part of it is about good practice (avoiding extra code in class files) but here it's much worse than that. With weatherDataSource.js file looking like this, **merely importing WeatherDataSource class will result with unwanted interaction with DOM triggered by this extra code**.

Reply:
This is only for testing, so I commented all this function in my code